Python/import.c

```diff
@@ -837,14 +837,18 @@ PyImport_AddModule(const char *name)
 static void
 remove_module(PyObject *name)
 {
+    PyObject *type, *value, *traceback;
+    PyErr_Fetch(&type, &value, &traceback);
     PyObject *modules = PyImport_GetModuleDict();
+    if (!PyMapping_HasKey(modules, name)) {
+        goto out;
```

Comment:
This code is simple and we can get rid of goto.

Reply:
I prefer the usage of goto to get a code looking like try/finally: have a single "finalize" block at the end. IMHO it makes the code more readable, and it's a common pattern in CPython.